### ISSUE TITLE
Unused triggering on Attributes

### ISSUE THREAD

**Issue by rdohms** [squizlabs/PHP_CodeSniffer#3452]
**Describe the bug**
When importing a class to be used as an attribute, PHPCS reports a `SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse` violation.
As it does not seem to detect the attribute usage as a valid usage.

It's clearly described here by @jrfnl https://github.com/rdohms/phpunit-arraysubset-asserts/pull/57

**Code sample**
```php
use ReturnTypeWillChange;

class ArrayAccessible implements ArrayAccess, IteratorAggregate
{
     *
     * @return mixed
     */
    #[ReturnTypeWillChange]
    public function offsetGet($offset)
    {
        return $this->array[$offset];
    }
{
```

**Custom ruleset**
```xml
<ruleset>
    <arg name="basepath" value="." />
    <arg name="extensions" value="php" /> <!-- which extensions to look for -->
    <arg name="parallel" value="80" /> <!-- how many parallel processes to run -->
    <arg name="colors" />
    <arg name="cache" value=".phpcs.cache" /> <!-- cache the results and don't commit them -->
    <arg value="np" /> <!-- n = ignore warnings, p = show progress -->

    <file>assertarraysubset-autoload.php</file>
    <file>src</file>
    <file>tests</file>

    <rule ref="DMS">
        <exclude name="SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingNativeTypeHint"/>
    </rule>

    <rule ref="SlevomatCodingStandard.Operators.DisallowEqualOperators.DisallowedEqualOperator">
        <exclude-pattern>src/Constraint/ArraySubset\.php</exclude-pattern>
    </rule>

    <!-- BC-layer: exclude rules which would prevent code from being PHP cross-version compatible. -->
    <rule ref="Squiz.Classes.ClassFileName.NoMatch">
        <exclude-pattern>assertarraysubset-autoload\.php</exclude-pattern>
        <exclude-pattern>src/ArraySubsetAssertsEmpty\.php</exclude-pattern>
        <exclude-pattern>src/AssertFallThrough\.php</exclude-pattern>
    </rule>
    <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes.DeclareStrictTypesMissing">
        <exclude-pattern>assertarraysubset-autoload\.php</exclude-pattern>
        <exclude-pattern>src/ArraySubsetAssertsEmpty\.php</exclude-pattern>
        <exclude-pattern>src/AssertFallThrough\.php</exclude-pattern>
        <exclude-pattern>tests/bootstrap\.php</exclude-pattern>
        <exclude-pattern>tests/Availability/*\.php</exclude-pattern>
    </rule>
    <rule ref="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedName">
        <exclude-pattern>assertarraysubset-autoload\.php</exclude-pattern>
        <exclude-pattern>tests/Availability/*\.php</exclude-pattern>
    </rule>
    <rule ref="SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint">
        <exclude-pattern>assertarraysubset-autoload\.php</exclude-pattern>
        <exclude-pattern>src/AssertFallThrough\.php</exclude-pattern>
    </rule>
    <rule ref="SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint">
        <exclude-pattern>assertarraysubset-autoload\.php</exclude-pattern>
        <exclude-pattern>src/AssertFallThrough\.php</exclude-pattern>
        <exclude-pattern>tests/Availability/*\.php</exclude-pattern>
    </rule>

</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above.
2. Run `phpcs test.php ...`
3. See error message displayed
```
FILE: test.php
------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------
 9 | ERROR | [x] Type ReturnTypeWillChange is not used in this file.
   |       |     (SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse)
------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------------------
```

**Expected behavior**
No errors should be reported.

**Versions (please complete the following information):**
 - PHP: 8.0
 - PHPCS: 3.6.1

**Comment by rdohms** [comment]
Derp, that explains why i could not find a report similar... my bad.
